leptos_dom/src/node_ref.rs

```diff
@@ -4,38 +4,76 @@ use leptos_reactive::{
 };
 use std::cell::Cell;
 
-/// Contains a shared reference to a DOM node created while using the `view`
-/// macro to create your UI.
-///
-/// ```
-/// # use leptos::{*, logging::log};
-/// use leptos::html::Input;
-///
-/// #[component]
-/// pub fn MyComponent() -> impl IntoView {
-///     let input_ref = create_node_ref::<Input>();
-///
-///     let on_click = move |_| {
-///         let node =
-///             input_ref.get().expect("input_ref should be loaded by now");
-///         // `node` is strongly typed
-///         // it is dereferenced to an `HtmlInputElement` automatically
-///         log!("value is {:?}", node.value())
-///     };
-///
-///     view! {
-///       <div>
-///       // `node_ref` loads the input
-///       <input _ref=input_ref type="text"/>
-///       // the button consumes it
-///       <button on:click=on_click>"Click me"</button>
-///       </div>
-///     }
-/// }
-/// ```
-#[repr(transparent)]
-pub struct NodeRef<T: ElementDescriptor + 'static>(
-    RwSignal<Option<HtmlElement<T>>>,
+cfg_if::cfg_if!(
```

Comment:
The `cfg_if` block and duplicated code can be replaced with
```rust
#[cfg(debug_assertions)]
defined_at: &'static std::panic::Location<'static>,
```
in the struct definition and 
```rust
#[cfg(debug_assertions)]
defined_at,
```
when it's used.

Reply:
@martinfrances107 Any chance you can or have made this change so we can merge this?